indexer/services/roundtable/__tests__/tasks/update-wallet-total-volume.test.ts

```diff
@@ -0,0 +1,150 @@
+      [],
+      {},
+    );
+
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      subaccountId: defaultSubaccountId[0].id,
+      createdAt: referenceDt.toISO(),
+      eventId: testConstants.defaultTendermintEventId,
+      price: '1',
+      size: '1',
+    });
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      subaccountId: defaultSubaccountId[0].id,
+      createdAt: referenceDt.plus({ years: 1 }).toISO(),
+      eventId: testConstants.defaultTendermintEventId2,
+      price: '2',
+      size: '2',
+    });
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      subaccountId: defaultSubaccountId[0].id,
+      createdAt: referenceDt.plus({ years: 2 }).toISO(),
+      eventId: testConstants.defaultTendermintEventId3,
+      price: '3',
+      size: '3',
+    });
+
+    await walletTotalVolumeUpdateTask();
+
+    const wallet = await WalletTable.findById(testConstants.defaultWallet.address);
+    expect(wallet).toEqual(expect.objectContaining({
+      ...testConstants.defaultWallet,
+      totalVolume: '14', // 1 + 4 + 9
+    }));
+  });
+
+  it('Succeeds in incremental totalVolume updates', async () => {
+    const defaultSubaccountId = await SubaccountTable.findAll(
+      { subaccountNumber: testConstants.defaultSubaccount.subaccountNumber },
+      [],
+      {},
+    );
+
+    // First task run: one new fill
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      subaccountId: defaultSubaccountId[0].id,
+      createdAt: DateTime.utc().toISO(),
+      eventId: testConstants.defaultTendermintEventId,
+      price: '1',
+      size: '1',
+    });
+    await walletTotalVolumeUpdateTask();
+    let wallet = await WalletTable.findById(testConstants.defaultWallet.address);
+    expect(wallet).toEqual(expect.objectContaining({
+      ...testConstants.defaultWallet,
+      totalVolume: '1',
+    }));
+
+    // Second task run: no new fills
+    await walletTotalVolumeUpdateTask();
+    wallet = await WalletTable.findById(testConstants.defaultWallet.address);
+    expect(wallet).toEqual(expect.objectContaining({
+      ...testConstants.defaultWallet,
+      totalVolume: '1',
+    }));
+
+    // Third task run: one new fill
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      subaccountId: defaultSubaccountId[0].id,
+      createdAt: DateTime.utc().toISO(),
+      eventId: testConstants.defaultTendermintEventId2,
+      price: '1',
+      size: '1',
+    });
+    await walletTotalVolumeUpdateTask();
+    wallet = await WalletTable.findById(testConstants.defaultWallet.address);
+    expect(wallet).toEqual(expect.objectContaining({
+      ...testConstants.defaultWallet,
+      totalVolume: '2',
+    }));
+  });
+
+  it('Successfully updates totalVolumeUpdateTime in persistent cache table', async () => {
+    await walletTotalVolumeUpdateTask();
+    const lastUpdateTime1 = await getTotalVolumeUpdateTime();
+
+    // Sleep for 1s
```

Comment:
why sleep here?

Reply:
To create a significant inverval between runs in case time difference is less than smallest precision in postgres which is microseconds. I dont think our code is that fast, so I can remove if preferred.